content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -568,6 +568,43 @@ The new Job itself will have a different uid from `a8f3d00d-c6d2-11e5-9f87-42010
 `manualSelector: true` tells the system to that you know what you are doing and to allow this
 mismatch.
 
+### Job tracking with finalizers
+
+{{< feature-state for_k8s_version="v1.22" state="alpha" >}}
+
+{{< note >}}
+In order to use this behavior, you must enable the `JobTrackingWithFinalizers`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/)
+and the [controller manager](/docs/reference/command-line-tools-reference/kube-controller-manager/).
+It is disabled by default.
+
+When enabled, the Job controller tracks new Jobs using the behavior described
+below. Existing Jobs are unaffected. Users don't interact with this feature.
+{{< /note >}}
+
+Traditionally, the Job controller uses the Pods that exist in the cluster to
+track the Job status, that is, to keep the counters for `succeeded` and `failed`
+Pods. However, Pods can be removed for a number of reasons, including:
+- The garbage collector that removes orphan Pods when a Node goes down.
+- The garbage collector that removes finished Pods (in `Succeeded` or `Failed`
+  phase) after a threshold.
+- The action of a user or a non-core controller.
+
+The new algorithm ensures that the Job status is up to date before a Pod is
+removed from the API server. To do that, the Job controller creates Pods with
+the finalizer `batch.kubernetes.io/job-tracking`. The controller removes the
```

Comment:
We'll soon have a glossary entry for finalizer, I hope. Mentioning this here as a tip for PR #28840

Reply:
Replaced that PR with #28870 because I accidentally pushed the Hugo themes folder and...yeah :(